tensorflow/lite/micro/micro_allocation_info.cc

```diff
@@ -204,6 +204,10 @@ TfLiteStatus AllocationInfoBuilder::InitializeAllocationInfo(
           (current->bytes != 0);
       if (offline_offsets) {
         current->offline_offset = offline_offsets[i];
+        if (subgraph->tensors()->Get(i)->is_variable() &&
```

Comment:
can we add a comment explaining this? it seems subtle.

Reply:
We need to set needs_allocating here only for offline planned variable tensors and not for all tensors. Added a comment.